src/parameterizations/lateral/MOM_tidal_forcing.F90

```diff
@@ -358,30 +345,18 @@ subroutine tidal_forcing_init(Time, G, US, param_file, CS)
     return
   endif
 
-  call get_param(param_file, mdl, "TIDAL_SAL_FROM_FILE", CS%tidal_sal_from_file, &
+  call get_param(param_file, mdl, "TIDAL_SAL_FROM_FILE", CS%use_tidal_sal_file, &
                  "If true, read the tidal self-attraction and loading "//&
                  "from input files, specified by TIDAL_INPUT_FILE. "//&
                  "This is only used if TIDES is true.", default=.false.)
-  call get_param(param_file, mdl, "USE_PREVIOUS_TIDES", CS%use_prev_tides, &
+  call get_param(param_file, mdl, "USE_PREVIOUS_TIDES", CS%use_tidal_sal_prev, &
                  "If true, use the SAL from the previous iteration of the "//&
                  "tides to facilitate convergent iteration. "//&
                  "This is only used if TIDES is true.", default=.false.)
-  call get_param(param_file, mdl, "TIDE_USE_SAL_SCALAR", CS%use_sal_scalar, &
-                 "If true and TIDES is true, use the scalar approximation "//&
-                 "when calculating self-attraction and loading.", &
-                 default=.not.CS%tidal_sal_from_file)
   ! If it is being used, sal_scalar MUST be specified in param_file.
-  if (CS%use_sal_scalar .or. CS%use_prev_tides) &
-    call get_param(param_file, mdl, "TIDE_SAL_SCALAR_VALUE", CS%sal_scalar, &
-                 "The constant of proportionality between sea surface "//&
-                 "height (really it should be bottom pressure) anomalies "//&
-                 "and bottom geopotential anomalies. This is only used if "//&
-                 "TIDES and TIDE_USE_SAL_SCALAR are true.", units="m m-1", &
-                 fail_if_missing=.true.)
-
-  call get_param(param_file, mdl, "TIDAL_SAL_SHT", CS%tidal_sal_sht, &
-                 "If true, use the online spherical harmonics method to calculate "//&
-                 "self-attraction and loading term in tides.", default=.false.)
+  if (CS%use_tidal_sal_prev) &
+    call get_param(param_file, mdl, "SAL_SCALAR_VALUE", CS%sal_scalar, fail_if_missing=.true., &
```

Comment:
The logic for setting SAL_SCALAR_VALUE should probably be revised to mirror what is done in SAL_init, although the fail_if_missing flag would have to be removed if we are setting a default value, as is done there.